src/components/Checkbox.tsx

```diff
@@ -0,0 +1,131 @@
+import type {
+  CheckboxProps as RTCheckboxProps,
+  ToggleProps,
+} from '@react-types/checkbox';
+import type { ComponentPropsWithRef, ReactNode } from 'react';
+import { useState, useRef } from 'react';
+import { useCheckbox } from 'react-aria';
+import { useToggleState } from 'react-stately';
+import { styled } from '../../stitches.config';
+import { CheckIcon } from '../animationIcons/CheckIcon';
+import { easeOutExpo } from '../primitives/animation';
+
+const Label = styled('label', {
+  '& input:checked:disabled + div': {
```

Comment:
What are we needing to target specifically with `div` here? Can we tighten this to reference IconContainer specifically?